internal/core/application/operator_service.go

```diff
@@ -826,75 +827,147 @@ func (o *operatorService) FeeAccountBalance(ctx context.Context) (
 	return int64(baseAssetAmount), nil
 }
 
+type depositType string
+
+var (
+	marketDeposit depositType = "MarketDeposit"
+	feeDeposit    depositType = "FeeDeposit"
+)
+
 func (o *operatorService) ClaimMarketDeposit(
 	ctx context.Context,
 	market Market,
 	outpoints []TxOutpoint,
 ) error {
-	if market.BaseAsset != config.GetString(config.BaseAssetKey) {
-		return domain.ErrInvalidBaseAsset
-	}
+	addressesPerAccount := make(map[int][]string)
+	bkPairs := make([][]byte, 0)
+	var accountIndex int
+	var err error
 
-	_, accountIndex, err := o.marketRepository.GetMarketByAsset(
-		ctx,
-		market.QuoteAsset,
-	)
-	if err != nil {
-		return err
-	}
+	existingMarketReq := existingMarketReq(market)
+	if existingMarketReq {
+		if err := validateAssetString(market.BaseAsset); err != nil {
+			return domain.ErrInvalidBaseAsset
+		}
 
-	addresses, bkPairs, err := o.vaultRepository.
-		GetAllDerivedAddressesAndBlindingKeysForAccount(
+		if err := validateAssetString(market.QuoteAsset); err != nil {
+			return domain.ErrInvalidQuoteAsset
+		}
+
+		if market.BaseAsset != config.GetString(config.BaseAssetKey) {
```

Comment:
Just a notice: In the #301 we are decoupling config package from app services, will leave it for now, but will eventually need to be rebased after we do the change in that PR